language/src/regent/optimize_index_launches.t

```diff
@@ -247,10 +263,15 @@ local function analyze_noninterference_previous(
         -- Index non-interference is handled at the type checker level
         -- and is captured in the constraints.
     then
-      return false, i
+      table.insert(args_interfering, i)
     end
   end
-  return true
+
+  if #args_interfering == 0 then
```

Comment:
I'm not sure `#` can be trusted if you're going to potentially have `nil` entries in the list. See: https://www.lua.org/manual/5.1/manual.html#2.5.5

I think you actually need to iterate this with `pairs` and break out of the loop if you find something.

Reply:
I replaced it with `next(args_interefering) == nil` which might be more efficient.